sdk/program/src/vote/state/mod.rs

```diff
@@ -347,16 +374,38 @@ impl VoteState {
         3762 // see test_vote_state_size_of.
     }
 
-    #[allow(clippy::used_underscore_binding)]
-    pub fn deserialize(_input: &[u8]) -> Result<Self, InstructionError> {
+    // we preserve the old deserialize on not(target_os = "solana") because the new does not support V0_23_5
+    pub fn deserialize(input: &[u8]) -> Result<Self, InstructionError> {
         #[cfg(not(target_os = "solana"))]
         {
-            deserialize::<VoteStateVersions>(_input)
+            deserialize::<VoteStateVersions>(input)
                 .map(|versioned| versioned.convert_to_current())
                 .map_err(|_| InstructionError::InvalidAccountData)
         }
         #[cfg(target_os = "solana")]
-        unimplemented!();
+        {
+            let mut vote_state = Self::default();
+            Self::deserialize_into(input, &mut vote_state)?;
+            Ok(vote_state)
+        }
+    }
+
+    pub fn deserialize_into(
+        input: &[u8],
+        vote_state: &mut VoteState,
+    ) -> Result<(), InstructionError> {
```

Comment:
micro-nit: Could you add a doc comment to this function explaining when to use it? Something like:

```
Deserializes the input buffer into the provided `VoteState`

This function exists to deserialize `VoteState` in a BPF context without going above
the stack frame limit, and must be kept up to date with `bincode::deserialize`.
```

Or feel free to update that as you wish!